### ISSUE TITLE
Is there a way through hexo to import existing blog content from a github repo or do I have to version control my hexo workspace folder?

### ISSUE THREAD

**Issue by EMChamp** [hexojs/hexo#1878]
Sorry if this is a bit of a beginner question I am not sure where else to post this. 

What is the process to clone a hexo blog workspace? I have created a blog on my local machine and it is up and running on github pages. Then I want to 'git clone' my hexo blog on Cloud9 IDE so I can work on it anywhere. 

```
git clone git@github.com:EMChamp/emchamp.github.io.git'

cd emchamp.github.io/'
```

My blog content is in the date folders /2016/04/08 so I know the content made it over. I continue with the installation.

```
npm install

//to generate hexo files
hexo init
```

My static files are still fine but the source/ folder is just a fresh instance. I'm pretty sure here is where I am messing up and my source/ folder should reflect my static files. Is my only option to 'git clone' my actual project folder or is there a way in hexo to import my blog content from github to the source/ (and other) folders?

```
npm install hexo-deployer-git

//Point to github repo
vim _config.yml

hexo generate --deploy
```

Now my static files and dynamic files are wiped out and that of a new repo which is expected behavior. 

TL;DR Is there a way through hexo to import existing blog content from a github repo or do I have to version control my hexo workspace folder?

**Comment by leesei** [comment]
I think you checked in the wrong content to master.

https://github.com/leesei/hexo-vanilla-site/tree/master
This is the Hexo site's source which contains the md and theme.
You `npm install` and `hexo <cmd>` here.

https://github.com/leesei/hexo-vanilla-site/tree/gh-pages
This is the branch `hexo deploy` push to.
The content is actually `public/` Hexo site's source (created upon `hexo generate`).

GitHub repos are only used as storage, think of the different branches as remote folders.
There's nothing magical about it.

**Comment by aratic** [comment]
Hello there, 

i had the same question, that i would like to "continue a blog", but lost my local copy since i'm now using a different pc,

the "published repo for storage", in my case, https://github.com/aratic/aratic.github.io , folder structure / layout is different with the one that "hexo init" would create, 

was it the case, that i should actually keep the local copy of blogging to a different repo, says the raw _post files ? and think .github.io page as a "write only" place just for generated static files ?

thanks for your time

**Comment by aratic** [comment]
after do "hexo init" and "hexo generate", one could clone / copy the name.github.io repo, put all content to the ./public folder of the newly created hexo folder, before doing the "hexo deploy". bad part is, one still lose the _post source, so ideally, we may need to do our own version control
